python/cugraph/cugraph/link_prediction/wsorensen.py

```diff
@@ -0,0 +1,107 @@
+    the user. Sorensen coefficient is defined between two sets as the ratio of
+    twice the volume of their intersection divided by the volume of each set.
+
+    Parameters
+    ----------
+    graph : cugraph.Graph
+        cuGraph graph descriptor, should contain the connectivity information
+        as an edge list (edge weights are not used for this algorithm). The
+        adjacency list will be computed if not already present.
+
+    weights : cudf.DataFrame
+        Specifies the weights to be used for each vertex.
+        Vertex should be represented by multiple columns for multi-column
+        vertices.
+
+        weights['vertex'] : cudf.Series
+            Contains the vertex identifiers
+        weights['weight'] : cudf.Series
+            Contains the weights of vertices
+
+    vertex_pair : cudf.DataFrame
+        A GPU dataframe consisting of two columns representing pairs of
+        vertices. If provided, the sorensen coefficient is computed for the
+        given vertex pairs, else, it is computed for all vertex pairs.
+
+    Returns
+    -------
+    df : cudf.DataFrame
+        GPU data frame of size E (the default) or the size of the given pairs
+        (first, second) containing the Sorensen weights. The ordering is
+        relative to the adjacency list, or that given by the specified vertex
+        pairs.
+
+        df['source'] : cudf.Series
+            The source vertex ID
+        df['destination'] : cudf.Series
+            The destination vertex ID
+        df['sorensen_coeff'] : cudf.Series
+            The computed weighted Sorensen coefficient between the source and
+            destination vertices.
+
+    Examples
+    --------
+    >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
+    >>>                   dtype=['int32', 'int32', 'float32'], header=None)
+    >>> G = cugraph.Graph()
+    >>> G.from_cudf_edgelist(M, source='0', destination='1')
+    >>> df = cugraph.sorensen_w(G, M['2'])
+    """
+    if type(input_graph) is not Graph:
+        raise Exception("input graph must be undirected")
+
+    if type(vertex_pair) == cudf.DataFrame:
+        vertex_pair = renumber_vertex_pair(input_graph, vertex_pair)
+    elif vertex_pair is None:
+        pass
+    else:
+        raise ValueError("vertex_pair must be a cudf dataframe")
+
+    if input_graph.renumbered:
+        vertex_size = input_graph.vertex_column_size()
+        if vertex_size == 1:
+            weights = input_graph.add_internal_vertex_id(
+                weights, 'vertex', 'vertex'
+            )
+        else:
+            cols = weights.columns[:vertex_size].to_list()
+            weights = input_graph.add_internal_vertex_id(
+                weights, 'vertex', cols
+            )
+    jaccard_weights = cudf.Series(np.ones(len(weights)))
+    for i in range(len(weights)):
+        jaccard_weights[weights['vertex'].iloc[i]] = weights['weight'].iloc[i]
```

Comment:
This piece of code was removed because we can directly copy `weights['weight']` to jaccard_weights. No need to create a cudf.Series of `np.ones` and iteratively modify its value. I also updated `wjaccard.py` since that was also present there